web/src/main/java/com/epam/esm/web/filter/JwtTokenFilter.java

```diff
@@ -0,0 +1,72 @@
+package com.epam.esm.web.filter;
+
+import com.epam.esm.service.service.user.UserService;
+import com.epam.esm.web.security.auth.Account;
+import com.epam.esm.web.security.jwt.JwtTokenUtil;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+
+@Component
+public class JwtTokenFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private final JwtTokenUtil jwtTokenUtil;
+
+    @Autowired
+    private final UserService userService;
+
+    @Autowired
+    public JwtTokenFilter(JwtTokenUtil jwtTokenUtil, UserService userService) {
+        this.jwtTokenUtil = jwtTokenUtil;
+        this.userService = userService;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response,
+                                    FilterChain chain) throws ServletException, IOException {
+
+        final String header = request.getHeader(HttpHeaders.AUTHORIZATION);
+        if (StringUtils.isEmpty(header) || !header.startsWith("Bearer ")) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        final String token = header.split(" ")[1].trim();
+            if (!jwtTokenUtil.validate(token)) {
+                chain.doFilter(request, response);
+            }
+
+            UserDetails userDetails = userService.getByLogin(jwtTokenUtil.getUsername(token))
+                    .map(Account::new)
+                    .orElseThrow(() -> new UsernameNotFoundException("User not found!"));
+
+            UsernamePasswordAuthenticationToken authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null,
+                    Optional.ofNullable(userDetails).map(UserDetails::getAuthorities).orElse(List.of())
```

Comment:
Looks like it can be only:
UsernamePasswordAuthenticationToken authentication = new UsernamePasswordAuthenticationToken(
                userDetails, null, userDetails.getAuthorities());